### DISCUSSION TITLE
Get List/Library Specific Items (large data)

### DISCUSSION THREAD

**Discussion by MichelAlHayek** [pnp/pnpcore#1333] [General]
What is the quickest and most performant way to load item(s) in a list/library that contains more than 10,000 items/files? I would like also to filter the library based on the _dlc_docId field values and to load more than 20 SP Fields for each document.

**Comment by kfircs** [comment]
Hi @MichelAlHayek, 

**_My_** first choice will be using paging with **Caml Query**, but this is really depends on many parameters - such as you **row limit**, **field types** and so on, and also can be affected on **tenant** or **geo location** traffic, if it is on the **cloud\onprem**.
so to summarize I'd suggest to go ahead and **benchmark** it depends on your specific needs there are many tools that can assist you with benchmark the different requests.
you can also look for benchmarks online for different ways to load items\files.

on top of it, if those items\files doesn't need to be immediately retrievable and can wait for the crawler to pick them up - you can also try to use search.
